sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -764,3 +760,188 @@ def update_repository_properties(self, *args, **kwargs):
+            headers={"Accept": OCI_MANIFEST_MEDIA_TYPE},
+            cls=_return_response_header,
+            **kwargs
+        )
+        
+        digest = response['Docker-Content-Digest']
+        if not _validate_digest(data, digest):
+            raise HttpResponseError("The digest in the response does not match the digest of the uploaded manifest.")
+        
+        return digest
+
+    @distributed_trace
+    def upload_blob(self, repository, data, **kwargs):
+        # type: (str, IO, **Any) -> str
+        """Upload an artifact blob.
+
+        :param str repository: Name of the repository
+        :param data: The blob to upload.
+        :type data: IO
+        :returns: The digest of the uploaded blob, calculated by the registry.
+        :rtype: str
+        :raises ValueError: If the parameter repository or data is None.
+        """
+        if repository is None or data is None:
+            raise ValueError("The parameter repository and data cannot be None.")
+
+        start_upload_response = self._client.container_registry_blob.start_upload(
+            repository, cls=_return_response_header, **kwargs
+        )
+        upload_chunk_response = self._client.container_registry_blob.upload_chunk(
+            start_upload_response['Location'], data, cls=_return_response_header, **kwargs
+        )
+        digest = _compute_digest(data)
+        complete_upload_response = self._client.container_registry_blob.complete_upload(
+            digest=digest, next_link=upload_chunk_response['Location'], cls=_return_response_header, **kwargs
+        )
+        return complete_upload_response['Docker-Content-Digest']
+
+    @distributed_trace
+    def download_manifest(self, repository, tag_or_digest, **kwargs):
+        # type: (str, str, **Any) -> DownloadManifestResult
+        """Download the manifest for an OCI artifact.
+
+        :param str repository: Name of the repository
+        :param str tag_or_digest: The tag or digest of the manifest to download.
+        :returns: DownloadManifestResult
+        :rtype: ~azure.containerregistry.models.DownloadManifestResult
+        :raises ValueError: If the parameter repository or tag_or_digest is None.
+        :raises ~azure.core.exceptions.HttpResponseError:
+            If the requested digest does not match the digest of the received manifest.
+        """
+        if repository is None or tag_or_digest is None:
+            raise ValueError("The parameter repository and tag_or_digest cannot be None.")
+
+        response = self._client.container_registry.get_manifest(
+            name=repository,
+            reference=tag_or_digest,
+            accept=OCI_MANIFEST_MEDIA_TYPE,
+            headers={"Accept": OCI_MANIFEST_MEDIA_TYPE},
+            cls=_return_response,
+            **kwargs
+        )
+        digest = response.http_response.headers['Docker-Content-Digest']
+        content = response.http_response.internal_response._content
```

Comment:
This is going to break if the transport is swapped out. We should not use the `internal_response` object at all - only the `http_response`.